### ISSUE TITLE
NaN's during training and unable to detect objects.

### ISSUE THREAD

**Issue by PaulKMandal** [pjreddie/darknet#1560]
Hello everyone. Long time reader, first time poster. When I attempt to train `yolov3-tiny` on my own dataset, the neural network is unable to train. Here is an example of how my labels are formatted: 

`1 0.469270833333 0.802777777778 0.240625 0.353703703704`

Attached is an image of the corresponding picture.
![VID_20190409_115651_003](https://user-images.githubusercontent.com/40133940/56236113-b1934100-603d-11e9-82a5-f6675594e37e.jpeg)

Here is the output I get when I attempt to train the neural network:
![Screenshot_2019-04-14_13-10-01](https://user-images.githubusercontent.com/40133940/56236214-e901ed80-603d-11e9-9189-b255534d0f82.png)

I am a bit confused whether I need to change the input layer of the NN so that the pixels match my images. When I test the pretrained NN, the sizes of the images never seem to matter (and I thought that's why we convert the labels from pixel values to decimal values).

Thank you for your help. I'm using YOLO for a project that is due soon so the sooner your responses the better.

**Comment by PaulKMandal** [comment]
I have changed the input layer to match my images and am recalculating the anchors. I will try to train the neural network either today or tomorrow.

**Comment by PaulKMandal** [comment]
Have another update. I changed the input from 416 x 416 to the size of my training images and recalculated the anchors. This led to the following error:
```
double free or corruption (out)
Aborted (core dumped)
```

 I then changed the input back to default and recalculated the anchors using the build in `calc_anchors` command in the AlexyAB version of darknet. The train command will work again, but the neural network still isn't training. Here is the command I use to calculate the weights:

```
./darknet detector calc_anchors cfg/obj.data -num_of_clusters 6 -width 416 -height 416 -show

 num_of_clusters = 6, width = 416, height = 416 
 read labels from 542 images 
 loaded 	 image: 542 	 box: 542
 all loaded. 

 calculating k-means++ ...

 iterations = 8 


 avg IoU = 89.99 % 

Saving anchors to the file: anchors.txt 
anchors =  94,100,  64,164,  90,137,  81,221, 113,193,  97,231
```
Once again, your input is greatly appreciated.

**Comment by PaulKMandal** [comment]
Okay, loss is getting low (I get 0.000778), but all of the outputs still list -nan.
![Screenshot_2019-04-20_15-36-56](https://user-images.githubusercontent.com/40133940/56463077-55ecee80-6382-11e9-8f63-ea5e7b9fe896.png)

Also, does it list 51840 images because it puts the images through different filters? I only have 600 images and am training on 540 of them.

**Comment by carrytameng** [comment]
Do you have any ideas now? I have encountered similar problems.

**Comment by PaulKMandal** [comment]
@carrytameng I'm still working on it.

**Comment by carrytameng** [comment]
@PaulKMandal I get it done by changing the image file from jpeg to jpg. refer issue: [1233](https://github.com/pjreddie/darknet/issues/1233)

**Comment by PaulKMandal** [comment]
@carrytameng I'll try that when I'm working at my workstation later today, thanks.

**Comment by PaulKMandal** [comment]
@carrytameng That solved the -nan's, but I am still having issues with detection. When I run the darknet detector test command, a black box appears.

**Comment by PaulKMandal** [comment]
I'm closing the issue. Thank you very much for your help. Everything is working smoothly.

**Comment by khiemubuntu** [comment]
you need learning_rate = 0,001 step = burn_in = 200

**Comment by theemtan** [comment]
@PaulKMandal How did you change your image type from .jpeg to .jpg? Did you just renamed the files? Cause I have .jpg files but still same results, showing -nan.
